src/pool/srv_pool.c

```diff
@@ -2631,6 +2632,21 @@ ds_pool_prop_set_handler(crt_rpc_t *rpc)
 	if (rc != 0)
 		D_GOTO(out_lock, rc);
 
+	/* Read all props & update prop IV */
+	rc = pool_prop_read(&tx, svc, DAOS_PO_QUERY_PROP_ALL, &prop);
+	if (rc != 0) {
+		D_ERROR(DF_UUID": failed to read prop for pool, rc=%d\n",
+			DP_UUID(in->psi_op.pi_uuid), rc);
+		D_GOTO(out_lock, rc);
+	}
+	D_ASSERT(prop != NULL);
+
+	rc = pool_iv_prop_update(svc->ps_pool, prop);
```

Comment:
seems better to do pool_iv_prop_update() after unlock and rdb_tx_end(), but that possibly cause new prop_update overwritten by old update and inconsistent on different nodes.
Not very sure if it is good to do iv_update/bcast when hold ps_lock, may see comment from @liw

Reply:
probably, I just followed what you did in ds_pool_connect_handler(). :)

Reply:
right, I remember it, but seems that in pool_connect (operation allowed after connect) it is with less impact than prop_set_handler. maybe fine for now.

Reply:
pool_iv_prop_update currently uses CRT_IV_SYNC_EAGER. So calling it while holding the pool metadata lock might be undesirable, as Xuezhao pointed out.

Reply:
My concern of moving it out of metadata lock is that we could end up in inconsistent pool props on different targets. (as Xuezhao mentioned above), any thoughts?

Reply:
Yeah, feels like versions are the way to go.